<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="image_view">
       <attribute name="title">
        <string>Image</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SetMaximumSize</enum>
          </property>
          <property name="leftMargin">
           <number>10</number>
          </property>
          <property name="topMargin">
           <number>10</number>
          </property>
          <property name="rightMargin">
           <number>10</number>
          </property>
          <property name="bottomMargin">
           <number>10</number>
          </property>
          <item>
           <widget class="QWidget" name="matplotlibEmbedWidget" native="true">
            <property name="minimumSize">
             <size>
              <width>480</width>
              <height>400</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <property name="spacing">
             <number>10</number>
            </property>
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>R:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="r_value"/>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>G:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="g_value"/>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>B:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="b_value"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPushButton" name="update_btn">
              <property name="text">
               <string>Update</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="histogram_view">
       <attribute name="title">
        <string>Histogram</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QWidget" name="histogramWidget" native="true"/>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="minimumSize">
             <size>
              <width>160</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>300</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="title">
             <string>Histogram Picker</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="1" column="0">
              <spacer name="verticalSpacer_2">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>40</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="0">
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
               <item>
                <widget class="QCheckBox" name="r_hist_checkbox">
                 <property name="text">
                  <string>R Value</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="g_hist_checkbox">
                 <property name="text">
                  <string>G Value</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="b_hist_checkbox">
                 <property name="text">
                  <string>B Value</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="avg_hist_checkbox">
                 <property name="text">
                  <string>Average</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHistogram">
    <property name="title">
     <string>Histogram</string>
    </property>
    <widget class="QMenu" name="menuEqualize">
     <property name="title">
      <string>Equalize</string>
     </property>
     <addaction name="actionEqualizeGrayscale"/>
     <addaction name="actionEqualizeYCrCb"/>
    </widget>
    <addaction name="actionNormalize"/>
    <addaction name="menuEqualize"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionBrightness"/>
    <addaction name="actionGrayscale"/>
   </widget>
   <widget class="QMenu" name="menuBinarization">
    <property name="title">
     <string>Binarization</string>
    </property>
    <addaction name="actionOtsu"/>
    <addaction name="actionNiblack"/>
    <addaction name="actionBinary_Thresholding"/>
   </widget>
   <widget class="QMenu" name="menuFilter">
    <property name="title">
     <string>Filter</string>
    </property>
    <widget class="QMenu" name="menuLow_Pass_Filter">
     <property name="title">
      <string>Low Pass Filters</string>
     </property>
     <addaction name="actionBox_Blur"/>
     <addaction name="actionGaussian_Blur"/>
    </widget>
    <addaction name="actionLinear_Filter"/>
    <addaction name="actionKuwahara_Filter"/>
    <addaction name="actionMedian_Filter"/>
    <addaction name="menuLow_Pass_Filter"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHistogram"/>
   <addaction name="menuBinarization"/>
   <addaction name="menuFilter"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionNormalize">
   <property name="text">
    <string>Normalize</string>
   </property>
  </action>
  <action name="actionBrightness">
   <property name="text">
    <string>Brightness</string>
   </property>
  </action>
  <action name="actionEqualizeGrayscale">
   <property name="text">
    <string>Grayscale</string>
   </property>
  </action>
  <action name="actionEqualizeYCrCb">
   <property name="text">
    <string>YCrCb</string>
   </property>
  </action>
  <action name="actionDefaultBW">
   <property name="text">
    <string>Default</string>
   </property>
  </action>
  <action name="actionRed_ChannelBW">
   <property name="text">
    <string>Red Channel</string>
   </property>
  </action>
  <action name="actionGreen_ChannelBW">
   <property name="text">
    <string>Green Channel</string>
   </property>
  </action>
  <action name="actionBlue_ChannelBW">
   <property name="text">
    <string>Blue Channel</string>
   </property>
  </action>
  <action name="actionOtsu">
   <property name="text">
    <string>Otsu</string>
   </property>
  </action>
  <action name="actionNiblack">
   <property name="text">
    <string>Niblack</string>
   </property>
  </action>
  <action name="actionBinary_Thresholding">
   <property name="text">
    <string>Binary Thresholding</string>
   </property>
  </action>
  <action name="actionAverageBW">
   <property name="text">
    <string>Average</string>
   </property>
  </action>
  <action name="actionGrayscale">
   <property name="text">
    <string>Grayscale</string>
   </property>
  </action>
  <action name="actionLinear_Filter">
   <property name="text">
    <string>Linear Filters</string>
   </property>
  </action>
  <action name="actionKuwahara_Filter">
   <property name="text">
    <string>Kuwahara</string>
   </property>
  </action>
  <action name="actionMedian_Filter">
   <property name="text">
    <string>Median Filter</string>
   </property>
  </action>
  <action name="actionBox_Blur">
   <property name="text">
    <string>Box Blur</string>
   </property>
  </action>
  <action name="actionGaussian_Blur">
   <property name="text">
    <string>Gaussian Blur</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
